hw/meson.build

```diff
@@ -0,0 +1,20 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Reggen invocation for generating the register headers we use in the software
+# to ensure register offsets continue to match the hardware.
+gen_hw_hdr_output = '@BASENAME@_regs.h'
```

Comment:
Can you manipulate this to be a function of `@INPUT@` such that path hierarchy where the output is created is replicated? `regtool` could be updated to create the directory if it does not exist.